src/legacy/ui/public/url/url.js

```diff
@@ -121,12 +119,7 @@ export function KbnUrlProvider($injector, $location, $rootScope, $parse, Private
           }));
       }
 
-      // append uriescape filter if not included
-      if (expr.indexOf('uriescape') === -1) {
-        expr += '|uriescape';
-      }
-
-      return $parse(expr)(paramObj);
+      return encodeURIComponent($parse(expr)(paramObj));
```

Comment:
Can you explain why we always need to encode this?

Reply:
A. It's always best to encode URL params
B. It's the previous behavior :)